src/Microsoft.ML.Tokenizers/Model/BertTokenizer.cs

```diff
@@ -0,0 +1,729 @@
+            }
+
+            if (tokenIds1 is not null)
+            {
+                for (int i = 0; i < tokenIds1.Count() + 1; i++) // Add 1 for [SEP] token.
+                {
+                    buffer[written++] = 1;
+                }
+            }
+
+            return OperationStatus.Done;
+        }
+
+        /// <summary>
+        /// Create a new instance of the <see cref="BertTokenizer"/> class.
+        /// </summary>
+        /// <param name="vocabFilePath">The path to the vocabulary file.</param>
+        /// <param name="doLowerCase">A value indicating whether the tokenizer should lowercase the input text.</param>
+        /// <param name="doBasicTokenization">A value indicating whether the tokenizer should do basic tokenization. Like clean text, normalize it, lowercasing, etc.</param>
+        /// <param name="splitOnSpecialTokens">A value indicating whether the tokenizer should split on special tokens.</param>
+        /// <param name="unknownToken">The unknown token. A token that is not in the vocabulary cannot be converted to an ID and is set to be this token instead.</param>
+        /// <param name="sepToken">The separator token, which is used when building a sequence from multiple sequences, e.g. two sequences for sequence classification or for a text and a question for question answering. It is also used as the last token of a sequence built with special tokens.</param>
+        /// <param name="padToken">The token used for padding, for example when batching sequences of different lengths.</param>
+        /// <param name="clsToken">The classifier token which is used when doing sequence classification (classification of the whole sequence instead of per-token classification). It is the first token of the sequence when built with special tokens.</param>
+        /// <param name="maskToken">The token used for masking values. This is the token used when training this model with masked language modeling. This is the token which the model will try to predict.</param>
+        /// <param name="tokenizeChineseChars">A value indicating whether the tokenizer should split the Chinese characters into tokens.</param>
+        /// <param name="stripAccents">A value indicating whether the tokenizer should strip accents characters.</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static BertTokenizer Create(
+                    string vocabFilePath,
+                    bool doLowerCase = true,
+                    bool doBasicTokenization = true,
+                    bool splitOnSpecialTokens = true,
+                    string unknownToken = "[UNK]",
+                    string sepToken = "[SEP]",
+                    string padToken = "[PAD]",
+                    string clsToken = "[CLS]",
+                    string maskToken = "[MASK]",
+                    bool tokenizeChineseChars = true,
+                    bool stripAccents = false) =>
```

Comment:
This is a lot of optional parameters. Do we expect there to ever be more? I'm wondering if we should have an options bag type.